src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -355,6 +289,23 @@ final class BlockingOutputBuffer(blockSize: Int, out: OutputBlockBuffer) extends
     fromRegion.loadBytes(fromOff, buf, off, n)
     off += n
   }
+
+  def writeDoubles(from: Array[Double], fromOff0: Int, n0: Int) {
+    assert(n0 >= 0)
```

Comment:
also assert `fromOff0 >= 0` and `n0 < from.length - fromOff0`

(or `n0 + fromOff0 < from.length` but the LHS could overflow. on the other hand, I don't see subtraction or toLong used elsewhere protectively (e.g. in `Region`) so maybe you think it's overkill?